crates/core/src/transport/connection_handler.rs

```diff
@@ -358,23 +380,26 @@ impl UdpPacketsListener {
                             if let SymmetricMessagePayload::AckConnection { result: Ok(_) } =
                                 packet.payload
                             {
-                                return Ok(ConnectionInfo {
-                                    outbound_symmetric_key: outbound_sym_key
-                                        .expect("should be set at this stage"),
-                                    inbound_symmetric_key: inbound_sym_key,
-                                    remote_public_key,
-                                    remote_is_gateway: false,
-                                    remote_addr,
-                                });
+                                let (inbound_sender, inbound_recv) = mpsc::channel(1);
+                                return Ok((
+                                    RemoteConnection {
+                                        outbound_symmetric_key: outbound_sym_key
+                                            .expect("should be set at this stage"),
+                                        inbound_symmetric_key: inbound_sym_key,
```

Comment:
Thinking further, we should store the entire intro packet rather than relying on a hash - because a hash would require reading the entire packet to determine whether it is the same as the intro packet - while if we store the intro packet then a non-match will just require reading the first few bytes.